src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -135,7 +141,7 @@ public BoundExpression VisitBinaryOperator(BoundBinaryOperator node, BoundUnaryO
             for (BoundBinaryOperator? current = node; current != null && current.ConstantValueOpt == null; current = current.Left as BoundBinaryOperator)
             {
                 // The regular visit mechanism will handle this.
-                if (current.InterpolatedStringHandlerData is not null || current.OperatorKind is BinaryOperatorKind.Utf8Addition)
+                if (current.InterpolatedStringHandlerData is not null || current.OperatorKind is BinaryOperatorKind.Utf8Addition || IsBinaryStringConcatenation(current))
```

Comment:
> || IsBinaryStringConcatenation(current) [](http://example.com/codeflow?start=128&length=40)

Do we have a test observing significance of this condition?

Reply:
Yes. `StringConcatTests.ConcatOneArgWithNullToString` will go through this path and hit one of the `throw ExceptionUtilities.UnexpectedValue()`s below.